File: app/src/main/java/com/trofiventures/testchameleonframework/service/ChameleonApi.java
package com.trofiventures.testchameleonframework.service;

import com.trofiventures.testchameleonframework.model.SkinClient;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;

/**
 * only for test
 */
public interface ChameleonApi {

    @GET("skin/{skin_id}")
    Call<Object> getSkin(@Header("FINTV-Auth-Token") String token,
                         @Path("skin_id") String def);

    @POST("skin/client")
    Call<Object> getSkin(@Body SkinClient skinClient);

}

File: chameleon/src/main/java/com/trofiventures/chameleon/service/ChameleonApi.kt
package com.trofiventures.chameleon.service

import com.trofiventures.chameleon.model.SkinClient
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

/**
 * only for test
 */
interface ChameleonApi {

    @POST("skin/v2/getSkin")
    fun getSkin(@Body skinClient: SkinClient): Call<Any>
}
